puppet/zulip/files/postgresql/process_fts_updates

```diff
@@ -1,24 +1,11 @@
-#!/usr/bin/env python3
+#!/home/zulip/deployments/current/zulip-py3-venv/bin/python3
 # Processes updates to postgres Full Text Search for new/edited messages.
 #
 # Zulip manages its postgres full-text search as follows.  When the
 # content of a message is modified, a postgres trigger logs the
 # message ID to the `fts_update_log` table.  In the background, this
 # program processes `fts_update_log`, updating the postgres full-text
 # search column search_tsvector in the main zerver_message.
-import sys
-
-# We want to use a virtualenv in production, which will be in /home/zulip/deployments/current.
-# So we should add that path to sys.path and then import scripts.lib.setup_path_on_import.
-# But this file is also used in development, where the above path will not exist.
-# So `import scripts.lib.setup_path_on_import` will raise an ImportError.
-# In development, we just want to skip this step since we know that virtualenv will already be in use.
-# So catch the ImportError and do nothing.
-sys.path.append('/home/zulip/deployments/current')
-try:
-    import scripts.lib.setup_path_on_import
-except ImportError:
-    pass
```

Comment:
This puppet tool is unique and is going to be broken.  This script needs to support running both with the development virtualenv (if available) or without (if it's running on a dedicated postgres server, as we do in production).

Reply:
I suppose we could address development by just changing the invocation in `tools/run-dev.py`.  

For production, there's sorta this awkward setup where we want to import the Django settings if available, but not if they aren't, which could probably be hacked around using `get-django-setting` or something?

Reply:
> I suppose we could address development by just changing the invocation in `tools/run-dev.py`.

Which is precisely what I did.

> For production, there's sorta this awkward setup where we want to import the Django settings if available, but not if they aren't,

That logic wasn’t changed, there’s still an `except ImportError` for that.

But if this needs to run in production when there’s no virtualenv at all, we need to do something else. In that case: is the “We want to use a virtualenv in production” comment wrong? Where does `psychopg2` come from?

Reply:
[Oh](https://github.com/zulip/zulip/blob/2.0.4/puppet/zulip/manifests/process_fts_updates.pp#L6).

Okay, how about we leave this as `#!/usr/bin/env python3` and run it with the virtualenv interpreter in puppet/zulip/files/supervisor/conf.d/zulip_db.conf? Although that seems silly for a program being installed into `/usr/local/bin`—is it also invoked manually?

Reply:
It's only invoked from supervisord in normal operation; I could imagine running it directly when debugging (have done this once in the last few years, about 2 months ago); so just needs to be possible, not super handy, to do.